Description: Upstream use a remapped version of asm3 to ensure
 that it does not conflict with asm2 elsewhere in the Jenkins
 dependency tree.
 .
 In Debian/Ubuntu we have a consistent version in the tree (unlike
 upstream) so we don't need todo this.
Author: James Page <james.page@ubuntu.com>
Forwarded: not-needed

--- a/core/src/main/java/hudson/util/SubClassGenerator.java
+++ b/core/src/main/java/hudson/util/SubClassGenerator.java
@@ -25,13 +25,13 @@ package hudson.util;
 
 import hudson.PluginManager.UberClassLoader;
 import jenkins.model.Jenkins;
-import org.kohsuke.asm3.ClassWriter;
-import org.kohsuke.asm3.MethodVisitor;
-import org.kohsuke.asm3.Type;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
 
 import java.lang.reflect.Constructor;
 
-import static org.kohsuke.asm3.Opcodes.*;
+import static org.objectweb.asm.Opcodes.*;
 
 /**
  * Generates a new class that just defines constructors into the super types.
--- a/core/pom.xml
+++ b/core/pom.xml
@@ -584,6 +584,13 @@ THE SOFTWARE.
     </dependency>
 
     <dependency>
+      <!-- with this, stapler can load parameter names from the debug info -->
+      <groupId>asm</groupId>
+      <artifactId>asm-commons</artifactId>
+      <version>2.2.3</version>
+    </dependency>
+
+    <dependency>
       <groupId>org.kohsuke</groupId>
       <artifactId>access-modifier-annotation</artifactId>
       <version>1.0</version>
